src/python/pants/backend/jvm/targets/scala_library.py

```diff
@@ -45,7 +56,46 @@ def __init__(self, java_sources=None, payload=None, **kwargs):
     payload.add_fields({
       'java_sources': PrimitiveField(self.assert_list(java_sources, key_arg='java_sources')),
     })
-    super().__init__(payload=payload, **kwargs)
+
+    if ScoveragePlatform.global_instance().get_options().enable_scoverage:
+      # Settings scalac_plugins
+      # Preventing instrumentation of generated targets or targets in Scoverage blacklist option.
+      if not self.skip_instrumentation(**kwargs):
+        if scalac_plugins:
+          scalac_plugins.append(SCOVERAGE)
+        else:
+          scalac_plugins = [SCOVERAGE]
+
+      # Setting scalac_plugin_args
+      if scalac_plugin_args:
+        scalac_plugin_args.update(
+          {
+            "scoverage": ["writeToClasspath:true",
+              f"dataDir:{Target.compute_target_id(kwargs['address'])}"]
+          })
+      else:
+        scalac_plugin_args = {
+          "scoverage": ["writeToClasspath:true",
+            f"dataDir:{Target.compute_target_id(kwargs['address'])}"]
+        }
+
+      # Setting compiler_option_sets
+      if compiler_option_sets:
+        list(compiler_option_sets).append(SCOVERAGE)
+      else:
+        compiler_option_sets = [SCOVERAGE]
+
+      super().__init__(payload=payload,
```

Comment:
Having two super calls is pretty weird, and in general this has resulted in a lot of nesting.

An alternative might be to do something like:
```
scalac_plugins = ... or []
scalac_plugin_args = ... or []
compiler_option_sets = ... or []
if ScoveragePlatform.global_instance().get_options().enable_scoverage:
  scalac_plugins.append(..)
  scalac_plugin_args.update(..)
  compiler_option_sets.update(..)
super(...)
```